### ISSUE TITLE
[SJ] Add trigger to limit max number of stickies within the volume.

### ISSUE THREAD

**Issue by hexaflexahexagon** [momentum-mod/game#1833]
### Is your feature request related to a problem?

Considering how we made changes from tf2 where the total number of stickies you can use at once are limited by HP, it makes sense to give control over this back to mappers in the form of a more sophisticated trigger. 

### Describe the solution you'd like

Add a new trigger (maybe named `trigger_stick_limit` in line with https://docs.momentum-mod.org/entity/trigger_stick_explosive/ ?) that has a KV for number of stickies available within it. Default should probably be 3, which is also the default when no trigger is present.

There should also be a HUD element that shows up with the number of available stickybombs when it is not at the default value. That hud element will then require a cvar to toggle, being put in settings, opening a doc page, etc.

I.E. with default behavior you `+attack` 4 times and the first stickybomb you fired will fizzle. If this trigger were added and the limit were changed to something like 4 then after using `+attack` 4 times no stickybombs would fizzle, and you could do a quad-stickyjump.

### Additional context

Split off from https://github.com/momentum-mod/game/issues/562 

This really does not need to worry about any other weapons because rockets and concs already have intentionally limited lifespans (collision or timers), so this added handle really doesn't do anything to them or provide useful gameplay.

**Comment by braem** [comment]
I believe the stickybomb count HUD already has the ability to display more than 3 stickies, so it'd just be a matter of hookin that up.

**Comment by braem** [comment]
Implemented as `trigger_stickybomb_limit` as repurposed the stickycount HUD as mentioned above.
Intentionally does not share a namespace with `trigger_stick_limit` as that is for sticking behaviour of all explosives, whereas this is a limit only applicable to stickybomblauncher.

**Comment by hexaflexahexagon** [comment]
is there any technical reason this is limited to 2 at minimum rather than 1 or even 0? it would be pretty useful if it could actually go that low for various different things

**Comment by braem** [comment]
> is there any technical reason this is limited to 2 at minimum rather than 1 or even 0? it would be pretty useful if it could actually go that low for various different things

No technical reason, just think it wouldn't be useful to go that low. I guess can bump to 1, but 0?

**Comment by hexaflexahexagon** [comment]
0 would be useful for making areas that instantly fizzle all active stickies and/or prevent the player from shooting in cleaner way than nonade (bcuz it applies based on position rather than what you hit). 1 would be more of a gimmick probably but no reason to exclude the possibility imo.

**Comment by braem** [comment]
> 0 would be useful for making areas that instantly fizzle all active stickies and/or prevent the player from shooting in cleaner way than nonade (bcuz it applies based on position rather than what you hit). 1 would be more of a gimmick probably but no reason to exclude the possibility imo.

There is an option on nonade to immediately fizzle a sticky on enter, so if you are in it and shoot it'll have the same effect; fizzling immediately. For preventing fire I think shotlimit would be best when that's implemented

**Comment by hexaflexahexagon** [comment]
> There is an option on nonade to immediately fizzle a sticky on enter, so if you are in it and shoot it'll have the same effect; fizzling immediately

this doesn't fizzle active stickies though. you can probably do that with some IO but this seems like a much cleaner approach, i just don't understand the reasoning for arbitrary limits even if there may be other ways to achieve the same results. at the very least it should be minimum of 1 so that you can have things like a downair where you cant place a full triple at the bottom (min 2 doesn't work because you can shoot 2 in the top area, then full charge 1 and walk off + shoot. entity can't cover the whole tube because you need 3 max to do a proper catch once you start doing downairs) but again if there's no technical reasons why it can't be 0 it be minimum of 0

**Comment by braem** [comment]
I see what you mean by active stickies yeah. I'll bump min to 0

**Comment by hexaflexahexagon** [comment]
Another thing I noticed is that if you set the number to below the minimum (I did 1 but same thing probably happens for negative numbers if min becomes 0) it defaults to the maximum aka 20. That felt kinda weird because you'd think it defaults to the min if you go under but if min is 0 you'd need to use a negative to trigger this and I think if you're entering a negative it's kinda expected to not really work. So probably nothing needs to change then I'd say but worth noting.
